data-model/src/main/scala/za/co/absa/enceladus/model/ConformedSchema.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.model
+
+import org.apache.spark.sql.types.StructField
+import za.co.absa.enceladus.model.conformanceRule._
+
+case class ConformedSchema(schema: List[StructField], dataset: Dataset) {
+  def hasField(field: String): Boolean = {
+    if (schema.exists(_.name == field)) true else {
+      val ss = dataset.conformance.find {
+        case MappingConformanceRule(_, _, _, _, _, _, outputColumn, additionalColumns, _, _, _) =>
+          outputColumn == field || additionalColumns.getOrElse(Map()).contains(field)
+        case SingleColumnConformanceRule(_, _, outputColumn, _, inputColumnAlias) =>
+          outputColumn == field || field == outputColumn + "." + inputColumnAlias
+        case DropConformanceRule(_, _, _) => false
```

Comment:
This line is redundant, right? Found it a bit confusing

Reply:
The point is that DropConformanceRule will be ignored initally at this point. Then if the field is found in a regular rule, a drop one will be looked for later on in order to establish if the final schema has the column or not

Reply:
Ah I see, I didn't realize that `DropConformanceRule` is a subtype of `ConformanceRule`. Makes sense now